syntax = "proto3";

package syft.grid.messages;

import "proto/core/common/common_object.proto";
import "proto/core/io/address.proto";

message CreateInitialSetUpMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  string name = 3;
  string email = 4;
  string password = 5;
  string domain_name = 6;
  double budget = 7;
  syft.core.io.Address reply_to = 8;
  bytes signing_key = 9;
}

message GetSetUpMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  syft.core.io.Address reply_to = 3;
}

message GetSetUpResponse {
  syft.core.common.UID msg_id = 1;
  int32 status_code = 2;
  string content = 3;
  syft.core.io.Address address = 4;
}

message UpdateSetupMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  string domain_name = 3;
  string description = 4;
  bool daa = 5;
  string contact = 6;
  bytes daa_document = 8;
  repeated string tags = 9;
  syft.core.io.Address reply_to = 10;
}

message UpdateSetupResponse {
  syft.core.common.UID msg_id = 1;
  string content = 3;
  syft.core.io.Address address = 4;
}
